src/data/list/basic.lean

```diff
@@ -1584,6 +1584,19 @@ begin
     exact lt_of_add_lt_add_left' (lt_of_le_of_lt h $ add_lt_add_right (lt_of_not_ge h') _) }
 end
 
+-- Several lemmas about sum/head/tail for `list ℕ`.
+-- These rely on the fact that `default ℕ = 0`.
+
+lemma head_add_tail_sum (L : list ℕ) : L.head + L.tail.sum = L.sum :=
+by { cases L, { simp, refl, }, { simp, }, }
+
+lemma head_le_sum (L : list ℕ) : L.head ≤ L.sum :=
+nat.le.intro (head_add_tail_sum L)
+
+@[simp]
```

Comment:
I'm not sure this is such a good simp lemma in general.  It duplicates the list `L`, and subtractions of natural numbers are awkward.

Reply:
Okay. I should have added a note that I was already dubious about this one myself.